### ISSUE TITLE
Pruning Convolutional Neural Networks for Resource Efficient Inference

### ISSUE THREAD

**Issue by joyhuang9473** [number9473/nn-algorithm#124]
# Pruning Convolutional Neural Networks for Resource Efficient Inference #

- Author: Pavlo Molchanov, Stephen Tyree, Tero Karras, Timo Aila, Jan Kautz
- Origin: https://arxiv.org/abs/1611.06440
- Related:
  - https://github.com/jacobgil/pytorch-pruning
  - https://jacobgil.github.io/deeplearning/pruning-deep-learning?from=singlemessage
  - https://github.com/handong1587/handong1587.github.io/blob/master/_posts/deep_learning/2015-10-09-acceleration-model-compression.md

**Comment by joyhuang9473** [comment]
Lisandro79/JetsonCaffe: Pruning Neural Networks

https://github.com/Lisandro79/JetsonCaffe/wiki/Pruning-Neural-Networks

**Comment by joyhuang9473** [comment]
PerforatedCNNs accelerate convolutional neural networks (CNNs) by skipping evaluation of the convolutional layers in some of the spatial positions.

https://github.com/mfigurnov/perforated-cnn-caffe

**Comment by joyhuang9473** [comment]
Add pruning possibilities at inner_product_layer

https://github.com/BVLC/caffe/pull/4294

**Comment by joyhuang9473** [comment]
Apply simple pruning on Caffemodel

https://github.com/garion9013/impl-pruning-caffemodel

**Comment by joyhuang9473** [comment]
TensorFlow implementation of "Iterative Pruning"

https://github.com/garion9013/impl-pruning-TF

**Comment by xizi** [comment]
Hi, joyhuang9473, is there a feasible code project for accelerate my vgg19 model which is trained on private dataset?

**Comment by ghost** [comment]
I'd be interested in anyone has done work on actually detecting **when** a pruned (simplified) model is less accurate... that is, if I present you with a full model and a simplified (albeit less accurate) model... can we first look at the input and determine if an input is well classified by the less accurate model (i.e. it's "easy" to classify) or that it's in a "difficult zone" where the less accurate model is subject to significantly less accuracy than the full model?

In other words - given an accurate and less accurate pair of models (where one was pruned from the other), can I tag an input to go confidently down the less accurate (although likely faster) pruned model or tag it to go to the full model?
